launcher/ui/dialogs/CreateShortcutDialog.cpp

```diff
@@ -0,0 +1,220 @@
+    MinecraftAccountPtr defaultAccount = accounts->defaultAccount();
+    if (accounts->count() <= 0) {
+        ui->overrideAccountCheckbox->setEnabled(false);
+    } else
+        for (int i = 0; i < accounts->count(); i++) {
+            MinecraftAccountPtr account = accounts->at(i);
+            auto profileLabel = account->profileName();
+            if (account->isInUse())
+                profileLabel = tr("%1 (in use)").arg(profileLabel);
+            auto face = account->getFace();
+            QIcon icon = face.isNull() ? APPLICATION->getThemedIcon("noaccount") : face;
+            ui->accountSelectionBox->addItem(profileLabel, account->profileName());
+            ui->accountSelectionBox->setItemIcon(i, icon);
+            if (defaultAccount == account)
+                ui->accountSelectionBox->setCurrentIndex(i);
+        }
+}
+
+CreateShortcutDialog::~CreateShortcutDialog()
+{
+    delete ui;
+}
+
+void CreateShortcutDialog::on_iconButton_clicked()
+{
+    IconPickerDialog dlg(this);
+    dlg.execWithSelection(InstIconKey);
+
+    if (dlg.result() == QDialog::Accepted) {
+        InstIconKey = dlg.selectedIconKey;
+        ui->iconButton->setIcon(APPLICATION->icons()->getIcon(InstIconKey));
+    }
+}
+
+void CreateShortcutDialog::on_overrideAccountCheckbox_stateChanged(int state)
+{
+    ui->accountOptionsGroup->setEnabled(state == Qt::Checked);
+}
+
+void CreateShortcutDialog::on_targetCheckbox_stateChanged(int state)
+{
+    ui->targetOptionsGroup->setEnabled(state == Qt::Checked);
+    ui->worldSelectionBox->setEnabled(ui->worldTarget->isChecked());
+    ui->serverAddressBox->setEnabled(ui->serverTarget->isChecked());
+    stateChanged();
+}
+
+void CreateShortcutDialog::on_worldTarget_toggled(bool checked)
+{
+    ui->worldSelectionBox->setEnabled(checked);
+    stateChanged();
+}
+
+void CreateShortcutDialog::on_serverTarget_toggled(bool checked)
+{
+    ui->serverAddressBox->setEnabled(checked);
+    stateChanged();
+}
+
+void CreateShortcutDialog::on_worldSelectionBox_currentIndexChanged(int index)
+{
+    stateChanged();
+}
+
+void CreateShortcutDialog::on_serverAddressBox_textChanged(const QString& text)
+{
+    stateChanged();
+}
+
+void CreateShortcutDialog::stateChanged()
+{
+    QString result = m_instance->name();
+    if (ui->targetCheckbox->isChecked()) {
+        if (ui->worldTarget->isChecked())
+            result = tr("%1 - %2").arg(result, ui->worldSelectionBox->currentData().toString());
+        else if (ui->serverTarget->isChecked())
+            result = tr("%1 - Server %2").arg(result, ui->serverAddressBox->text());
+    }
+    ui->instNameTextBox->setPlaceholderText(result);
+    if (!ui->targetCheckbox->isChecked())
+        ui->buttonBox->button(QDialogButtonBox::Ok)->setEnabled(true);
+    else
+        ui->buttonBox->button(QDialogButtonBox::Ok)
+            ->setEnabled(ui->worldTarget->isChecked() || (ui->serverTarget->isChecked() && !ui->serverAddressBox->text().isEmpty()));
```

Comment:
```suggestion
    else {
        ui->buttonBox->button(QDialogButtonBox::Ok)
            ->setEnabled((ui->worldTarget->isChecked() && ui->worldTarget->currentIndex() != -1) || (ui->serverTarget->isChecked() && !ui->serverAddressBox->text().isEmpty()));
    }
```

If you have no worlds it's buggy

Reply:
Good catch